applications/MappingApplication/custom_mappers/coupling_geometry_mapper.cpp

```diff
@@ -144,9 +144,9 @@ void CouplingGeometryMapper<TSparseSpace, TDenseSpace>::InitializeInterface(Krat
     // TODO Philipp I am pretty sure we should separate the vector construction from the matrix construction, should be independent otherwise no clue what is happening
     MappingMatrixUtilities::BuildMappingMatrix<TSparseSpace, TDenseSpace>(
         mpMappingMatrixSlave,
-        mpInterfaceVectorContainerOrigin->pGetVector(),
         mpInterfaceVectorContainerDestination->pGetVector(),
-        mpInterfaceVectorContainerOrigin->GetModelPart(),
+        mpInterfaceVectorContainerDestination->pGetVector(),
+        mpInterfaceVectorContainerDestination->GetModelPart(),
         mpInterfaceVectorContainerDestination->GetModelPart(),
         mMapperLocalSystemsSlave,
         echo_level);
```

Comment:
@philbucher the problem was mpMappingMatrixSlave was constructed with the wrong dims (it was rectangular like the projector, not square with [nDest x nDest]). Maybe a separate function in the mappingMatrixUtilities for mortar component matrices would be useful?

Reply:
awesome, nice catch thx

I will definitely this so that we can use it properly